schematic/schematics/redshift_schematic.py

```diff
@@ -0,0 +1,553 @@
+        return len(
+            str(value).encode("utf-8")) == len(
+            str(value)) and super(
+            RedshiftCharType,
+            self).is_value_compatible_with_class(value)
+
+
+class RedshiftBooleanType(schematic.TableColumnType):
+    """A boolean type in Redshift"""
+    name = "RedshiftBooleanType"
+    next_less_restrictive = RedshiftVarcharType
+    parameterized = False
+    valid_true_literals = ['TRUE', 't', 'true', 'y', 'yes', '1']
+    valid_false_literals = ['FALSE', 'f', 'false', 'n', 'no', '0']
+
+    def __init__(self):
+        super(RedshiftBooleanType, self).__init__()
+        self.len = len
+
+    def to_sql(self):
+        return "BOOLEAN".format(self.len)
+
+    def is_value_compatible_with_class(self, value):
+        """Determine if value can be inserted into column of
+           the group of types described by the class.
+
+        Args:
+          value: The value to check.
+        """
+        return value in self.valid_false_literals or value in self.valid_true_literals
+
+
+class RedshiftAbstractDatetimeType(schematic.TableColumnType):
+    """Abstract datetime type to provide subclasses compatibility
+    checking logic.
+
+    Attributes:
+      valid_regex: Any valid value for columns of the subclass's
+                   type will match this regex
+    """
+    valid_regex = None
+
+    def is_value_compatible_with_class(self, value):
+        """Check to see if a given value could be
+        inserted into a column of this type.
+
+        Args:
+          value: The value to check
+        Returns:
+          boolean indicating whether or not the value
+          is compatible
+        """
+        return bool(self.valid_regex.match(value))
+
+
+class RedshiftTimestampType(RedshiftAbstractDatetimeType):
+    """A timestamp type in Redshift"""
+    name = "RedshiftTimestampType"
+    next_less_restrictive = RedshiftVarcharType
+    parameterized = False
+    valid_regexes = [
+        "(^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|1[1-9]|2[1-9]|3[0-1])$)"]
+    valid_regex = re.compile("|".join(valid_regexes))
+
+    def __init__(self):
+        super(RedshiftTimestampType, self).__init__()
+
+    def to_sql(self):
+        return "TIMESTAMP".format(self.len)
+
+
+class RedshiftTimestampTZType(RedshiftAbstractDatetimeType):
+    """A Timestamp with time zone type in Redshift"""
+    name = "RedshiftTimestampTZType"
+    next_less_restrictive = RedshiftTimestampType
+    parameterized = False
+    valid_regexes = [
+        "(^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|1[1-9]|2[1-9]|3[0-1])$)"]
+    valid_regex = re.compile("|".join(valid_regexes))
+
+    def __init__(self):
+        super(RedshiftTimestampTZType, self).__init__()
+
+    def to_sql(self):
+        return "TIMESTAMPTZ".format(self.len)
+
+
+class RedshiftDateType(RedshiftAbstractDatetimeType):
+    """A DATE type in Redshift"""
+    name = "RedshiftDateType"
+    next_less_restrictive = RedshiftTimestampTZType
```

Comment:
```suggestion
    next_less_restrictive = RedshiftTimestampType
```
Assuming we agree that TimstampTZ is less restrictive than Timestamp